src/domains/chart/components/lib-charts/dygraph-chart.tsx

```diff
@@ -831,21 +831,29 @@ export const DygraphChart = ({
       // todo support state.tmp.dygraph_force_zoom
       const forceDateWindow = [viewAfter, viewBefore]
 
+      const dygraphOptionInEffect = dygraphInstance.current.getOption("dateWindow")
+      let hasChangedTimeframe = false
+      if (dygraphOptionInEffect.length > 0) {
+        // eslint-disable-next-line max-len
+        hasChangedTimeframe = Math.abs((viewBefore - viewAfter) - (dygraphOptionInEffect[1] - dygraphOptionInEffect[0])) > 5000
+      }
+
       // in old dashboard, when chart needed to reset internal dateWindow state,
       // dateWindow was set to null, and new dygraph got the new dateWindow from results.
       // this caused small unsync between dateWindow of parent (master) and child charts
       // i also detected that forceDateWindow timestamps have slightly better performance (10%)
       // so if the chart needs to change local dateWindow, we'll always use timestamps instead of
       // null.
 
-      const xAxisRange = dygraphInstance.current.xAxisRange()
+      const xAxisRange = dygraphInstance.current?.xAxisRange()
```

Comment:
why is that question mark here? It wasn't needed before

Reply:
this was introduced by the linter somehow. Fixed